bin/kaleidoscope-builder

```diff
@@ -58,23 +58,42 @@ build_filenames () {
 
 enable_ccache () {
     if  [ "$(command -v ccache)" ]; then
-    if ! [ -d "$CCACHE_WRAPPER_PATH" ]; then
-	mkdir -p "$CCACHE_WRAPPER_PATH"
-	ln -s "$(command -v ccache)" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}gcc"
-	ln -s "$(command -v ccache)" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}g++"
-	ln -s "${AVR_NM}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}nm"
-	ln -s "${AVR_OBJCOPY}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}objcopy"
-	ln -s "${AVR_AR}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}ar"
-	ln -s "${AVR_SIZE}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}size"
-    fi
-    export CCACHE_PATH=${COMPILER_PATH}/
-    CCACHE_ENABLE="-prefs compiler.path=${CCACHE_WRAPPER_PATH}/"
+      if ! [ -d "$CCACHE_WRAPPER_PATH" ]; then
+         mkdir -p "$CCACHE_WRAPPER_PATH"
+      fi
+      
+      if ! [ -h "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}gcc" ]; then
+         ln -s "$(command -v ccache)" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}gcc"
+      fi
+      
+      if ! [ -h "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}g++" ]; then
+         ln -s "$(command -v ccache)" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}g++"
+      fi
+      
+      if ! [ -h "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}nm" ]; then
+         ln -s "${AVR_NM}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}nm"
+      fi
+      
+      if ! [ -h "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}objcopy" ]; then
+         ln -s "${AVR_OBJCOPY}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}objcopy"
+      fi
+      
+      if ! [ -h "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}ar" ]; then
+         ln -s "${AVR_AR}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}ar"
+      fi
+      
+      if ! [ -h "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}size" ]; then
+         ln -s "${AVR_SIZE}" "${CCACHE_WRAPPER_PATH}/${COMPILER_PREFIX}size"
+      fi
+    
+      export CCACHE_PATH=${COMPILER_PATH}/
+      CCACHE_ENABLE="-prefs compiler.path=${CCACHE_WRAPPER_PATH}/"
     fi
 }
 
 
 firmware_size () {
-    if [ "${BOARD}" = "virtual" ]; then
+    if [ "${ARCH}" = "x86" ]; then
```

Comment:
Virtual builds are now triggered by defining `ARCH=x86`. The `BOARD` variable is left to specify the actual hardware that is supposed to be simulated.

Reply:
Can we call it something like x86_simulator or something? I can see some future where we have keyboard firmware running on an embedded x86 :)

Reply:
I'd even call the architecture `virtual` because who says that we are going to live with `x86` as host platform for ever. You point about `x86` becoming an embedded platform makes a lot of sense.